src/pages/JoinCirclePage/JoinCirclePage.tsx

```diff
@@ -53,43 +53,44 @@ export const JoinCirclePage = () => {
     profile?.users.some(u => u.circle_id === circleId);
 
   useEffect(() => {
-    assert(token);
-    const fn = async () => {
-      try {
-        const res = await fetch('/api/circle/landing/' + token);
+    try {
+      fetch('/api/circle/landing/' + token).then(res => {
         if (!res.ok) {
           setTokenError(
             'Invalid invite link; check with your Circle Admin for an updated link.'
           );
           return;
         }
-        const info: TokenJoinInfo = JSON.parse(await res.text());
+        res.json().then((info: TokenJoinInfo) => {
```

Comment:
Show there is a guild while this is loading. @CryptoGraffe

Reply:
added some simple loading text